### ISSUE TITLE
OroEntitySerializedFieldsBundle 4.1.5 does not work on Oro Platform 4.1

### ISSUE THREAD

**Issue by jimohalloran** [oroinc/platform#1064]
**Summary**  
Just did a composer update on an OroCRM 4.1 application.  Composer fetched and installed version 4.1.5 of `oro/platform-serialised-fields`.  

Subsequently attempting to run `oro:platform:update` results in the following error:

```
In OroEntityExtendBundle.php line 121:
                                                                                   
  In CheckExceptionOnInvalidReferenceBehaviorPass.php line 86:                     
                                                                                   
    The service "oro_serialized_fields.validator.extend_entity_serialized_data"    
     has a dependency on a non-existent service "oro_entity_config.validator.fi    
    eld_config_constraints_factory".                                               
```

It looks like 4.1.5 of `oro/platform-serialised-fields` introduces a dependency on a class (@oro_entity_config.validator.field_config_constraints_factory ) which does not exist in Oro Platform 4.1 (it wasn't introduced until 4.2).  

4.1.5 should have been released as a 4.2.* version, or the  oro/platform-serialised-fields dependency in crm-application updated to prevent 4.1.5 from being installed since it can't work with Oro Platform 4.1.

**Steps to reproduce**  
Run `composer update` on an otherwise working OroCRM 4.1.12 application.  Note that the `oro/platform-serialised-fields` dependency is updated.  Run `oro:platform:update` (or itterally any other Oro console command) and you'll see the error above.

**Actual Result**
See error message above.

**Expected Result**  
The update should complete successfully without requiring classes from Oro Platform 4.2.

**Details about your environment**
- OroPlatform version: 4.1.13
- PHP version: 7.3
- Database (MySQL, PostgreSQL) version: N/A

**Comment by chrisaligent** [comment]
We've just confirmed the same issue in OroCommerce 4.1 (Community Edition), running a `composer update` brings in `4.1.5` of the dependency:
![2021-06-03_11-34](https://user-images.githubusercontent.com/45612883/120575340-0d102b00-c460-11eb-92d0-a44ad32d1800.png)

This doesn't exist in 4.1:
![2021-06-03_11-37](https://user-images.githubusercontent.com/45612883/120575418-3630bb80-c460-11eb-82ec-f14ea001d5e5.png)

It seems to have been added to Platform 4.1.14 which is Enterprise-only.

**Comment by anyt** [comment]
Released v4.1.6 of oro/platform-serialised-fields that is actually 4.1.4 with the limitation in composer.json to "oro/platform": ">=4.1.0,<=4.1.13", so it should be in the latest version of a community edition of oro/platform v4.1
